problems/problems.go

```diff
@@ -15,6 +51,69 @@ func (t *DefaultDataProvider) GetSets() ([]byte, error) {
 	return ioutil.ReadFile(fileName)
 }
 
+// FindProblem retrieves problem details for a given user and problem ID. For now it only works with
+// Codeforces.
+func (t *DefaultDataProvider) FindProblem(userID string, problemID uint) (*Problem, error) {
+	problem, err := t.findProblem(problemID)
+	if err != nil {
+		return nil, err
+	}
+
+	// NOTE: Assuming this is a Codeforces problem for now
+	externalID := problem.ExternalID
+	contestID, err := strconv.Atoi(externalID[0 : len(externalID)-1])
+	index := externalID[len(externalID)-1:]
+
+	url := fmt.Sprintf(
+		"http://codeforces.com/api/contest.status?contestId=%d&handle=%s&showUnofficial=true",
+		contestID, userID,
+	)
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var submissionResponse CodeforcesSubmissionResponse
+	if err := json.Unmarshal(data, &submissionResponse); err != nil {
+		log.Printf("Error: %s", err)
+		return nil, err
+	}
+	for _, submission := range submissionResponse.Result {
+		if submission.Problem.Index == index {
+			problem.HasTried = true
+		}
+		if submission.Verdict == "OK" {
+			problem.HasSolved = true
+		}
+	}
+	return problem, nil
+}
+
+func (t *DefaultDataProvider) findProblem(problemID uint) (*Problem, error) {
+	fileName := os.Getenv("APP_ROOT") + "/problems/fixtures/problems.json"
+	data, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		return nil, err
+	}
+	var problems []Problem
+	if err := json.Unmarshal(data, &problems); err != nil {
+		return nil, err
+	}
+
+	for _, problem := range problems {
+		if problem.ID == problemID {
+			return &problem, nil
+		}
+	}
+	return nil, errors.New("problem not found")
```

Comment:
Minor, but the capitalization on some of this error messages is off.

Reply:
This one will be controversial, I used to capitalize these error messages, but `go lint` recommends not doing it: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
D=. I didn't know about that standard.  I don't disagree _that much_ as long as it is consistent across the code base.

Reply:
Sure, I'll try to fix the inconsistencies, because I have been capitalizing error messages, until I saw this warning.